src/librustc_typeck/check/mod.rs

```diff
@@ -2433,13 +2433,23 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     if arg_count == 1 {" was"} else {"s were"}),
                 error_code);
 
-            err.span_label(sp, format!("expected {}{} parameter{}",
-                                        if variadic {"at least "} else {""},
-                                        expected_count,
-                                        if expected_count == 1 {""} else {"s"}));
             if let Some(def_s) = def_span {
                 err.span_label(def_s, "defined here");
             }
+            if let Some(sugg) = sugg {
+                let mut sugg_span = sp.end_point();
+                // remove closing `)` from the span
+                sugg_span.hi = sugg_span.lo;
+                err.span_suggestion(
+                    sugg_span,
+                    "expected one unit argument. try creating a new unit value",
```

Comment:
this is a cute trick; is `unit` the terminology we are using here? somehow this message feels a bit "jargon-y" to me, but I'm not sure how to say it better.

cc @rust-lang/docs

Reply:
I like this one better, but the `;` looks out of place.

maybe like this?
```
help: expected a unit value. Try to insert one more pair of parenthesis?
```

Reply:
I updated the message